### ISSUE TITLE
Unexpected behavior for indent rule

### ISSUE THREAD

**Issue by calda** [nicklockwood/SwiftFormat#665]
I'm looking into adopting the `indent` rule in the [Airbnb Swift Style Guide](https://github.com/airbnb/swift). We have  an indentation rule (e.g. [_"Use 2 spaces to indent lines."_](https://github.com/airbnb/swift#spaces-over-tabs)) that is not currently being enforced with a SwiftFormat rule.

I'm seeing unexpected indentation behavior when I run `swiftformat . --rules indent --indent 2`:

```swift
// Before
self.method(
  withParameter: 1,
  otherParameter: 2) { [weak self] in
    guard let error = error else { return }
    print("and a trailing closure")
}

// After running `swiftformat . --rules indent --indent 2`
self.method(
  withParameter: 1,
  otherParameter: 2) { [weak self] in
  guard let error = error else { return } // I feel like the closure body should stay indented
  print("and a trailing closure")
}
```

Is this expected behavior of the `indent` rule, or is this a bug?

**Comment by nicklockwood** [comment]
This is deliberate. Indenting is relative to the starting of the expression where the scope begins, not the line where the scope begins.

I agree it looks odd in this example, but I would normally use the following style (which can also be enforced using the wrapArguments rule):

```swift

self.method(
  withParameter: 1,
  otherParameter: 2
) { [weak self] in
  guard let error = error else { return }
  print("and a trailing closure")
}
```

**Comment by calda** [comment]
I see -- we'd like to use [`--closingparen same-line`](https://github.com/airbnb/swift/issues/40) for `wrapArguments`, so I'm not sure if that suggestion would work for us.

Do you think it would be reasonable for us to contribute an option for this (like `--closurebody indent`)?

**Comment by nicklockwood** [comment]
> Do you think it would be reasonable for us to contribute an option for this (like --closurebody indent)?

I think there might be a wider application for such an option, e.g. for conditional statements with wrapped arguments:

```swift
if let foo = bar,
    let baz = quux {
        print("double-indented")
}
```

**Comment by keith** [comment]
Are you saying that would be the correct indent for that example? I would expect:

```swift
if let foo = bar,
    let baz = quux {
    print("double-indented")
}
```

**Comment by nicklockwood** [comment]
@keith I'm saying that if you wanted an option to indent function bodies in this way (which is what @calda was requesting IIUC) then you would probably also want to indent conditional blocks in the same way.

**Comment by calda** [comment]
This is how I would personally format multi-line `if` statements (and I'm fairly confident this would be our rule in the [Airbnb Swift Style Guide](https://github.com/airbnb/swift)):

```swift
if let foo = bar,
    let baz = quux 
{
    print("single-indented")
}
```

But I think we would prefer double-indenting trailing closure bodies (since this would match the indentation of non-trailing closure bodies):

```swift
self.method(
    withParameter: 1,
    otherParameter: 2) { [weak self] in
        guard let error = error else { return }
        print("and a trailing closure")
}
```

I think these two discussions are similar, but would be orthogonal configuration points (because we wouldn't want to use the same formatting behavior both of these cases).

**Comment by nicklockwood** [comment]
@calda this has landed in version 0.45.0. Thanks!

**Comment by calda** [comment]
Awesome!! Super excited that we can bring back some of our old Style Guide rules now
